Sources/TraceLog/Configuration.swift

```diff
@@ -0,0 +1,177 @@
+///
+///  Configuration.swift
+///
+///  Copyright 2016 Tony Stone
+///
+///  Licensed under the Apache License, Version 2.0 (the "License");
+///  you may not use this file except in compliance with the License.
+///  You may obtain a copy of the License at
+///
+///  http://www.apache.org/licenses/LICENSE-2.0
+///
+///  Unless required by applicable law or agreed to in writing, software
+///  distributed under the License is distributed on an "AS IS" BASIS,
+///  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///  See the License for the specific language governing permissions and
+///  limitations under the License.
+///
+///  Created by Tony Stone on 4/28/16.
+///
+import Foundation
+
+///
+/// Internal Configuration data
+///
+internal class Configuration {
+    
+    enum ConfigurationError : Error, CustomStringConvertible {
+        case invalidLogLevel(String)
+        
+        var description: String {
+            switch self {
+            case .invalidLogLevel(let value):
+                return value
+            }
+        }
+    }
+    
+    private static let logTag    = "LOG_TAG_"
+    private static let logPrefix = "LOG_PREFIX_"
+    private static let logAll    = "LOG_ALL"
+    
+    var globalLogLevel = LogLevel.info
+    
+    var loggedPrefixes: [String : LogLevel] = [:]
+    var loggedTags:     [String : LogLevel] = [:]
+    var writers:        [Writer]            = [ConsoleWriter()]
+    
+    init () {}
+    
+    ///
+    /// (Re)Load this structure with the values for writers and the environment variables
+    ///
+    func load(_ writers: [Writer], environment: Environment) -> [ConfigurationError] {
+        
+        var errors = [ConfigurationError]()
+        
+        self.globalLogLevel = LogLevel.info
+        
+        self.writers.removeAll()
+        self.loggedPrefixes.removeAll()
+        self.loggedTags.removeAll()
+
+        /// Initialize the writers first
+        self.writers.append(contentsOf: writers)
+        
+        for (variable, value) in environment {
+            ///
+            /// All variables and log level strings are converted
+            /// to upper case for comparison.
+            ///
+            let upperCaseVariable        = variable.uppercased()
+            let upperCaselogLevelString  = value.uppercased()
+            
+            if upperCaseVariable.hasPrefix(Configuration.logAll) {
+                
+                if let level = upperCaselogLevelString.asLogLevel() {
+                    self.globalLogLevel = level
+                } else {
+                    errors.append(.invalidLogLevel("Variable '\(upperCaseVariable)' has an invalid logLevel of '\(upperCaselogLevelString)'. '\(Configuration.logAll)' will be set to \(self.globalLogLevel)."))
```

Comment:
The test coming into this section is specifically for LOG_ALL (you can't enter this section unless that was passed).  The `self.globalLogLevel` is setup to a default value (currently INFO) at the beginning so if the `upperCaselogLevelString` is an invalid value, the following message evaluates to the proper message as in the tests (test message from `func testInitialize_LogWriters_Environment_GlobalInvalidLogLevel()` below).  

   "Variable 'LOG_ALL' has an invalid logLevel of 'TRACE5'. 'LOG_ALL' will be set to INFO."

I have changed the `'\(Configuration.logAll)'` to `'\(upperCaseVariable)'` statement to be more consistent with the rest of messages though.